qml/Panel/Indicators/IndicatorsLight.qml

```diff
@@ -91,55 +87,79 @@ QtObject {
         //   but 'caution' icon at 9%.
         // 
 
-        var lColor = ""
-        var lOnMS = -1
-        var lOffMS = -1
+        // only show led when display is off
+        if (displayStatus == Powerd.On) {
+            indicatorState = "INDICATOR_OFF"
+            return
+        }
 
-        var isCharging = batteryIconName.indexOf("charging") >= 0
-        if(!isCharging)
-            isCharging = deviceState != "discharging"
-
-        if(hasMessages) { 
-            // Unread Notifications
-            lColor  = "darkgreen"
-            lOnMS   = 1000
-            lOffMS  = 3000
-        } else if(isCharging) {
-            if(batteryIconName.indexOf("full-charged") >= 0
-               || deviceState == "fully-charged"
-               || batteryLevel >= 100) { 
-                // Battery Full
-                lColor  = "green"
-                lOnMS   = 1000
-                lOffMS  = 0
-            } else {
-                // Battery Charging
-                lColor  = "white"
-                lOnMS   = 1000
-                lOffMS  = 0
-            }
-        } else if(batteryIconName.indexOf("caution") >= 0
-                  || batteryIconName.indexOf("empty") >= 0) {
-            // Battery Low
-            lColor  = "orangered"
-            lOnMS   = 500
-            lOffMS  = 3000
+        // unread messsages have highest priority
+        if (hasMessages) {
+            indicatorState = "HAS_MESSAGES"
+            return
         }
 
-        console.log("  color=" + lColor + ", onMS=" + lOnMS + ", offMS=" + lOffMS)
-        if(lOnMS > -1) {
-            root.onMillisec = lOnMS
+        // show charging. show full only when charging
+        var isCharging = batteryIconName.indexOf("charging") >= 0
+                         || deviceState != "discharging"
+
+        var isFull = isCharging
+                     && (batteryIconName.indexOf("full-charged") >= 0
+                         || deviceState == "fully-charged"
+                         || batteryLevel >= 100)
+
+        if (isCharging) {
+            if (isFull)
```

Comment:
Due to the lines above `isFull` can only be true if also `isCharging` is true.
```
if (isFull) {
    indicatorState = "BATTERY_FULL"
} else if (isCharging) {
    indicatorState = "BATTERY_CHARGING"
} else if (batteryIconName.indexOf("caution") >= 0
           || batteryIconName.indexOf("empty") >= 0) {
    indicatorState = "BATTERY_LOW"
} else {
    indicatorState = "INDICATOR_OFF"
}

Reply:
That is intentional. I think the battery full indication must be turned off when the charger is disconnected.

Reply:
Yes, and I agree, but I'm saying that then the code can be simplify in the way I proposed.